generic/cxl.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Pooja B Surya <pooja@linux.vnet.ibm.com>
+
+import os
+
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import git
+from avocado.utils import build
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import distro
+
+
+class Cxl(Test):
+ 
+    def setUp(self):
+        # Check for basic utilities
+        lspci_out = process.system_output("lspci -v")
+        if "accelerators" not in lspci_out:
+            self.cancel("No capi card preset. Unable to initialte the test")          
+        sm = SoftwareManager()
+        deps = ['gcc', 'make', 'automake', 'autoconf']
+        for package in deps:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error(package + ' is needed for the test to be run')
+        git.get_repo('https://github.com/ibm-capi/cxl-tests.git',
+                     destination_dir=self.srcdir)
+        os.chdir(self.srcdir)
```

Comment:
Please refrain from using `srcdir`. It will get deprecated soon. `teststmpdir` will be preferred.